### PR TITLE
refactor(lidar_apollo_segmentation_tvm/_nodes): remove autoware_auto_common dependency

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Description
related to https://github.com/autowarefoundation/autoware.universe/issues/2229
remove autoware_auto_common from lidar_apollo_segmentation_tvm/_nodes

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by takayuki5168** [comment]
@yukke42 @yukkysaito 
I'm not familiar with these packages, but I wanna remove autoware_auto_common which is autoware.auto's common package.
For this reason, I created this PR. Could you check this PR please.

**Comment by yukkysaito** [comment]
@esteve Do you know the background and benefits of the introduction of these types?

**Comment by kenji-miyake** [comment]
Regarding this kind of code, let's refer to https://www.autosar.org/fileadmin/user_upload/standards/adaptive/21-11/AUTOSAR_RS_CPP14Guidelines.pdf and follow the latest recommendation.

https://autowarefoundation.github.io/autoware-documentation/main/contributing/coding-guidelines/languages/cpp/

**Comment by takayuki5168** [comment]
@kenji-miyake Could you give me more hint of how to refer to the guideline.

FYI: There is no behavior change with this PR since bool8_t, float32_t, float64_t is defined as bool, float, double respectively here.
https://github.com/autowarefoundation/autoware.universe/blob/7829c07d9a982054638312a44d4f6c374f8aa04d/common/autoware_auto_common/include/common/types.hpp#L39-L47

**Review by xmfcx — APPROVED** [review]

**Comment by xmfcx** [comment]
(Please ignore my review, it was a dummy test)

> @kenji-miyake Could you give me more hint of how to refer to the guideline. (-> Explained in a chat.)

@takayuki5168 I've opened up the following discussion to talk about it: https://github.com/orgs/autowarefoundation/discussions/3016

**Comment by ambroise-arm** [comment]
I feel like this PR is a step backwards if long term we want to indeed have fixed-width types.
Also, I don't understand the rational of removing the common package just because it is an autoware.auto package. It could simply be renamed.
And as for the "annoying dependencies" it creates, isn't it the case for any common package that aims at being used in several other packages in order to de-duplicate code?

In any case, those are not strongly held opinions and I don't oppose going ahead with the change.

**Comment by xmfcx** [comment]
@ambroise-arm 
> I don't understand the rational of removing the common package just because it is an autoware.auto package. It could simply be renamed.

It's not because it is an Autoware.Auto package, it's because it's a dependency. In https://github.com/orgs/autowarefoundation/discussions/3016 we've discussed if we should create a similar package for types in Autoware Core.

> I feel like this PR is a step backwards if long term we want to indeed have fixed-width types.

As discussed in https://github.com/orgs/autowarefoundation/discussions/3016 , here are some conclusions:
- Autoware doesn't require to be MISRA/AUTOSAR compliant in near future.
- We will encourage devs to use [`<cstdint>`](https://en.cppreference.com/w/cpp/header/cstdint) for integer types. And will try to add automatic checking features to CI to fix them automatically.
- When C++23 comes out and a ROS version supporting it comes out, we will use [`<stdfloat>`](https://en.cppreference.com/w/cpp/header/stdfloat)

As for `typedef bool8_t = bool` and `typedef char8_t = char` I think we can discuss it further.

> And as for the "annoying dependencies" it creates, isn't it the case for any common package that aims at being used in several other packages in order to de-duplicate code?

Since types are the most essential things, it needs to be included in every package. We will of course have these kind of things if they are necessary. But what we are discussing is about whether they are necessary or not for the current state of the Autoware, the users etc.

Please add your inputs to the [discussion](https://github.com/orgs/autowarefoundation/discussions/3016) as well.

**Comment by ambroise-arm** [comment]
> It's not because it is an Autoware.Auto package, it's because it's a dependency.

I may have misunderstood the meaning of the initial comment https://github.com/autowarefoundation/autoware.universe/pull/2234#issuecomment-1305691570. Alright then.

> We will encourage devs to use `<cstdint>` for integer types.

I had missed that the PR only replaced `floatXX_t` and `bool8_t`. Although it should also replace `char8_t ` by `char`. It probably won't compile in the current state (not picked up in CI as it doesn't compile the package by default).

**Comment by yukkysaito** [comment]
@takayuki5168 Can you resolve some conflicts?

**Comment by takayuki5168** [comment]
@yukkysaito @xmfcx I resolved the conflicts.

File: perception/lidar_apollo_segmentation_tvm/include/lidar_apollo_segmentation_tvm/cluster2d.hpp [hunk 1]
```diff
@@ -37,9 +36,8 @@ namespace perception
 {
 namespace lidar_apollo_segmentation_tvm
 {
-using autoware::common::types::bool8_t;
+
 using autoware::common::types::char8_t;
```
[the remark this answers is not in the document]
Comment (takayuki5168) [reply perception/lidar_apollo_segmentation_tvm/include/lidar_apollo_segmentation_tvm/cluster2d.hpp:39]: https://github.com/autowarefoundation/autoware.universe/pull/2234/commits/d7f9531f071714505f053d2b4231873f4d6b1cef
DONE

**Comment by yukkysaito** [comment]
LGTM

**Review by ambroise-arm — CHANGES_REQUESTED** [review]

**Comment by esteve** [comment]
> @esteve Do you know the background and benefits of the introduction of these types?

Sorry for the slow response, my notifications were not working properly. The reason for these types is mostly MISRA, because the built-in types don't have a defined size across all platforms, so by defining these aliases we can see quickly what size they have.

**Comment by xmfcx** [comment]
@takayuki5168 I've moved this branch to AWF organization to collaborate easier. And made some fixes. https://github.com/autowarefoundation/autoware.universe/pull/3136

In order to build it, you need to run `colcon build --symlink-install  --cmake-args -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_EXPORT_COMPILE_COMMANDS=1 -DDOWNLOAD_ARTIFACTS=1  --packages-select lidar_apollo_segmentation_tvm lidar_apollo_segmentation_tvm_nodes tvm_utility`

Without the ` --cmake-args -DDOWNLOAD_ARTIFACTS=1` argument, it will skip building the package.

**Comment by takayuki5168** [comment]
@xmfcx Sorry about not handling this PR. I really appreciate your help.

**Pull request closed without merging**